feat: add support for descriptions argument

dadangnh opened the pull request:
Hi Team, please kindly check the PR to add support for descriptions argument. Currently, without description variable, we need to re-create the address if we imported the address to terraform. With this PR, hopefully it solve the issue.

Related to #119

Thanks

apeabody commented:
/gcbrun

apeabody reviewed (changes requested):
Thanks for the contribution @dadangnh

From the linter:
```
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.ze5e9T5JoU/generate_docs/workspace/README.md
186a187
> | descriptions | A list of descriptions to add to each address. | `list(string)` | <pre>[<br>  ""<br>]</pre> | no |
206d206
< | descriptions | A list of descriptions to add to each address. | `string` | `""` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
```

dadangnh commented:
Thanks @apeabody , updated the PR, please kindly review

apeabody commented:
/gcbrun

dadangnh commented:
Hi @apeabody , updated the PR with the suggestion, please check again

apeabody commented:
/gcbrun

apeabody commented on variables.tf:
```diff
@@ -137,3 +137,9 @@ variable "labels" {
+variable "descriptions" {
+  description = "A list of descriptions to add to each address."
+  type        = list(string)
+  default     = [""]
```
Perhaps better to have the default list empty, rather than a single zero length string.

```suggestion
  default     = []
```

apeabody reviewed (changes requested):
Thanks for the contribution @dadangnh!

dadangnh commented:
Thanks for the review @apeabody , I updated as per your input. Please check again

apeabody commented:
> Thanks for the review @apeabody , I updated as per your input. Please check again

Thanks! - Looks like we just need to regenerate the docs:

```
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.aTe3Vprmt5/generate_docs/workspace/README.md
187c187
< | descriptions | A list of descriptions to add to each address. | `list(string)` | <pre>[<br>  ""<br>]</pre> | no |
---
> | descriptions | A list of descriptions to add to each address. | `list(string)` | `[]` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
```

dadangnh commented:
> > Thanks for the review @apeabody , I updated as per your input. Please check again
> 
> Thanks! - Looks like we just need to regenerate the docs:
> 
> ```
> Checking for documentation generation
> diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.aTe3Vprmt5/generate_docs/workspace/README.md
> 187c187
> < | descriptions | A list of descriptions to add to each address. | `list(string)` | <pre>[<br>  ""<br>]</pre> | no |
> ---
> > | descriptions | A list of descriptions to add to each address. | `list(string)` | `[]` | no |
> Error: Documentation generation has not been run, please run the
> 'make docker_generate_docs' command and commit the above changes.
> ```

Hi @apeabody , I've updated the README.md file with the `docker_generate_docs` command. Please review again.

apeabody commented:
/gcbrun

apeabody commented on main.tf:
```diff
@@ -58,6 +58,7 @@ resource "google_compute_address" "ip" {
   purpose      = var.address_type == "INTERNAL" ? var.purpose : null
   network_tier = var.address_type == "INTERNAL" ? null : var.network_tier
   labels       = var.labels
+  description  = element(var.descriptions, count.index)
```
If using `element()`, might need to wrap in `try(element(var.descriptions, count.index), null)`

```
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Error in function call
        	            	
        	            	  on ../../../main.tf line 61, in resource "google_compute_address" "ip":
        	            	  61:   description  = element(var.descriptions, count.index)
        	            	    ├────────────────
        	            	    │ while calling element(list, index)
        	            	    │ count.index is 0
        	            	    │ var.descriptions is empty list of string
        	            	
        	            	Call to function "element" failed: cannot use element function with an empty
        	            	list.}
        	Test:       	TestIpAddressOnly
```

dadangnh replied:
Oh my bad, yes, agree. I've updated the branch. Please run the lint check again.

apeabody commented:
/gcbrun

apeabody commented on main.tf:
```diff
@@ -70,6 +71,7 @@ resource "google_compute_global_address" "global_ip" {
   purpose       = var.global && var.address_type == "INTERNAL" ? "VPC_PEERING" : null
   prefix_length = local.prefix_length
   ip_version    = var.ip_version
+  description  = try(element(var.descriptions, count.index), null)
```
```suggestion
  description   = try(element(var.descriptions, count.index), null)
```

apeabody commented:
/gcbrun

apeabody reviewed (approved):
Thanks for the contribution @dadangnh!
